enterprise/views.py

```diff
@@ -248,10 +248,21 @@ def post_account_consent(self, request, consent_provided):
         Interpret the account-wide form above, and save it to a UserDataSharingConsentAudit object for later retrieval.
         """
         self.lift_quarantine(request)
+
         customer = get_enterprise_customer_for_request(request)
-        if customer is None:
+        social_auth = get_real_social_auth_object(request)
+        user = getattr(social_auth, 'user', None)
+        if customer is None or user is None:
```

Comment:
I'd prefer to see this broken up into separate guards because although the outcomes are similar the error cases are a bit different:

```
# Load the EnterpriseCustomer object tied to this request.  Trigger a 404 response if not available.
enteprise_customer = get_enterprise_customer_for_request(request)
if enterprise_customer is None:
    raise Http404

# Load the User object tied to the authentication object for this request.  Trigger a 404 response if not available.
social_auth = get_real_social_auth_object(request)
user = getattr(social_auth, 'user', None)
if user is None:
    raise Http404
```